### PR TITLE
dev-python/pytest-flask: new package, add 1.3.0

### PR DESCRIPTION
**Pull request by stkw0**
CC: @ArsenArsen 

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-python/pytest-flask/pytest-flask-1.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYPI_NO_NORMALIZE=1
+PYTHON_COMPAT=( python3_{11,12} pypy3 )
+inherit distutils-r1 pypi
+
+DESCRIPTION="A set of pytest fixtures to test Flask applications "
+HOMEPAGE="http://pytest-flask.readthedocs.org"
```
Comment (mgorny) [inline dev-python/pytest-flask/pytest-flask-1.3.0.ebuild:12]: Add GitHub and PyPI too.

File: dev-python/pytest-flask/pytest-flask-1.3.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="test"
```
Comment (mgorny) [inline dev-python/pytest-flask/pytest-flask-1.3.0.ebuild:17]: That's implied by `distutils_enable_tests`. If you really insist on making it explicit, you need to add `RESTRICT="!test? ( test )"` as well.

File: dev-python/pytest-flask/files/pytest-flask-fix-flask-compat.patch [hunk 3]
```diff
@@ -0,0 +1,74 @@
+diff --git a/pytest_flask/fixtures.py b/pytest_flask/fixtures.py
```
Comment (mgorny) [inline dev-python/pytest-flask/files/pytest-flask-fix-flask-compat.patch:1]: This patch isn't used.

**Review by mgorny — CHANGES_REQUESTED** [review]

File: dev-python/pytest-flask/metadata.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	</maintainer>
+	<maintainer type="person" proxied="yes">
+		<email>davidroman96@gmail.com</email>
+		<name>David Roman</name>
+	</maintainer>
+	<maintainer type="project" proxied="proxy">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<upstream>
```
Comment (mgorny) [inline dev-python/pytest-flask/metadata.xml:16]: Add python@ too (assuming tests pass).

File: dev-python/pytest-flask/pytest-flask-1.3.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYPI_NO_NORMALIZE=1
+PYTHON_COMPAT=( python3_{11,12} pypy3 )
+inherit distutils-r1 pypi
+
+DESCRIPTION="A set of pytest fixtures to test Flask applications "
```
Comment (mgorny) [inline dev-python/pytest-flask/pytest-flask-1.3.0.ebuild:11]:
```suggestion
DESCRIPTION="A set of pytest fixtures to test Flask applications"
```

File: dev-python/pytest-flask/pytest-flask-1.3.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="
+	dev-python/flask[${PYTHON_USEDEP}]
+	dev-python/werkzeug[${PYTHON_USEDEP}]
+"
+
+DEPEND="${RDEPEND}
```
Comment (mgorny) [inline dev-python/pytest-flask/pytest-flask-1.3.0.ebuild:27]:
```suggestion
BDEPEND="
	${RDEPEND}
```

File: dev-python/pytest-flask/pytest-flask-1.3.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+	test? (
+		dev-python/mock[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/pytest-flask/pytest-flask-1.3.0.ebuild:29]: They don't seem to be actually using it.

File: dev-python/pytest-flask/pytest-flask-1.3.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="
+	dev-python/flask[${PYTHON_USEDEP}]
+	dev-python/werkzeug[${PYTHON_USEDEP}]
+"
+
+DEPEND="${RDEPEND}"
```
Comment (mgorny) [inline dev-python/pytest-flask/pytest-flask-1.3.0.ebuild:27]: Do you actually need them to build the wheel?

Comment (stkw0) [reply dev-python/pytest-flask/pytest-flask-1.3.0.ebuild:27]: As far as I understand, DEPEND or BDEPEND were meant to be used to declare test dependencies. In this case as the tests also need the same libs than at runtime, it made sense to have DEPEND=$RDEPEND, but they don't seem to be strictly needed for building the package.

Comment (mgorny) [reply dev-python/pytest-flask/pytest-flask-1.3.0.ebuild:27]: That's `BDEPEND="test? ( ${RDEPEND} )"`, and `distutils_enable_tests` does it for you.

**Review by mgorny — APPROVED** [review]
Thanks!

**Comment by ArsenArsen** [comment]
thanks for the package (and reviews)!  applying shortly

**Pull request closed without merging**